src/Common/src/TypeSystem/Common/DefType.FieldLayout.cs

```diff
@@ -2,35 +2,81 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 
 namespace Internal.TypeSystem
 {
     // This is the api surface necessary to query the field layout of a type
-
     public abstract partial class DefType : TypeDesc
     {
+        private class FieldLayoutFlags
+        {
+            public const int HasContainsPointers = 1;
```

Comment:
Without comment it is not obvious the difference between HasContainsPointers and ContainsPointers. But when you look how the flags are used below, I think the HasXxxx flags should be renamed into IsXxxxComputed. It will nicely fit the pattern:

```
if (!_fieldLayoutFlags.HasFlags(FieldLayoutFlags.IsContainsPointersComputed)
{
    ComputeTypeContainsPointers();
}
```